### PR TITLE
Document the latest buildpack behavior

### PR DESCRIPTION
**Pull request by ericbottard**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,38 +3,63 @@ The riff Buildpack is a Cloud Native Buildpack V3 that provides riff Invokers to
...
+### Detection Phase
 Detection passes if 
 - a `$APPLICATION_ROOT/riff.toml` exists and 
 - Either
-  1. the build plan already contains a `jvm-application` key
-  2. as a fallback, the file pointed by the `artifact` value in `riff.toml` exists and is executable
+    1. the build plan already contains a `jvm-application` key (typically because a JVM based application was detected by the [java buildpack](https://github.com/cloudfoundry/build-system-buildpack))
+    2. the build plan already contains a `npm` key (typically because an NPM based application was detected by the [npm buildpack](https://github.com/cloudfoundry/npm-cnb))
+        
+        1. alternatively, if the file pointed by the artifact value in `riff.toml` exists and has a `.js` extension
```
Comment (trisberg) [inline README.md]: "pointed by"  sounds odd, maybe "referenced in" or "pointed to by"

File: README.md [hunk 2]
```diff
@@ -3,38 +3,63 @@ The riff Buildpack is a Cloud Native Buildpack V3 that provides riff Invokers to
...
 
 If a command function has been selected
-* Contributes the riff Command Invoker to a launch layer, set as the main executable with `FUNCTION_URI = <artifact>`
+* Contributes the riff Command Invoker to a launch layer, set as the main executable with `FUNCTION_URI = <artifact>` set as an environment variable.
 
 
 * Contributes `web` process
 * Contributes `function` process
 
+## In Plain English
```
Comment (jldec) [inline README.md]: i would suggest putting this section near the top, and and removing the first sentence below.

File: README.md [hunk 3]
```diff
@@ -3,38 +3,63 @@ The riff Buildpack is a Cloud Native Buildpack V3 that provides riff Invokers to
...
+If a node function has been detected
+* Contributes the riff Node Invoker to a launch layer, set as the main `node` entry point with `FUNCTION_URI = <artifact>` set as an environment variable.
+Note that `artifact` may actually be empty, in which case the invoker will `require()` the current directory (the function), which in turn expects that it contains a valid `package.json` file with its `main` entry point set. 
 
 If a command function has been selected
-* Contributes the riff Command Invoker to a launch layer, set as the main executable with `FUNCTION_URI = <artifact>`
+* Contributes the riff Command Invoker to a launch layer, set as the main executable with `FUNCTION_URI = <artifact>` set as an environment variable.
 
 
 * Contributes `web` process
```
Comment (jldec) [inline README.md]: could you add just a few words what these "process" things are? (or link to buildpacks docs)

File: README.md [hunk 4]
```diff
@@ -3,38 +3,63 @@ The riff Buildpack is a Cloud Native Buildpack V3 that provides riff Invokers to
...
 * Contributes `function` process
 
+## In Plain English
+The above two paragraphs describe the internal behavior of this buildpack. In a nutshell, when combined with the other buildpacks present in the [riff builder](https://github.com/projectriff/riff-buildpack-group) what this means (and especially when dealing with the riff CLI which takes care of the creation of the `riff.toml` file for you):
```
Comment (trisberg) [inline README.md]: should it be "two sections" instead of paragraphs?

**Comment by ericbottard** [comment]
Addressed review comments as seen in the second commit of this PR and rebased/squashed as 9f20370
